src/Core/Commander/ManagerCommands.js

```diff
@@ -71,43 +79,80 @@ define('Core/Commander/ManagerCommands', [
             return this.commandsLength()===0;
         };
 
-        ManagerCommands.prototype.runAllCommands = function() {
-
-
-            if (this.commandsLength() === 0)
-            {
-                return Promise.resolve(0);
-            }
-
-            return Promise.all(this.arrayDeQueue(16))
-                .then(function () {
-                    // if (this.commandsLength() <= 16)
-                    this.scene.wait(1);
-                    // else
-                    //     this.scene.renderScene3D();
-                    return this.runAllCommands();
-                }.bind(this));
+        ManagerCommands.prototype.resetExecutedCommandsCount = function() {
+            this.counters.executedCommands = 0;
         };
 
-        ManagerCommands.prototype.arrayDeQueue = function(number) {
-
-            var nT = number === undefined ? this.queueAsync.length : number;
-
-            var arrayTasks = [];
+        ManagerCommands.prototype.executeCommands = function() {
+
+            var command;
+            var launchCommand = function(cmd) {
+                this.counters.runningCommands++;
+                var layer = cmd.layer;
+                var host;
+                if(layer.url) {
+                    host = new URL(layer.url).host;
```

Comment:
Consider doing this operation just once when the command is created (in InterfaceCommander maybe?).